### PR TITLE
Make all of our JS libraries closure-warning free.

### PR DESCRIPTION
**Pull request by sbc100**
There are a couple of places where I think closure has bad/lacking
extern definitions upstream.

Otherwise this is moslut simply a questions of adding the needed type
annotation.  Hopefully now that I've addd a test for
INCLUDE_FULL_LIBRARY we won't see regressions going forward.

A few places there were actual unused arguments being passed.

Fixes: #12728

### CODE REVIEW

File: src/library.js [hunk 1]
```diff
@@ -1875,6 +1875,7 @@ LibraryManager.library = {
 
     return { family: family, addr: addr, port: port };
   },
+  $writeSockaddr__docs: '/** @param {number=} addrlen */',
```
Comment (kripken) [inline src/library.js:1878]: why are things like this needed? (what warning happens without it?)

Comment (sbc100) [reply src/library.js:1878]: closure treats all argument as required unless explicitly marked as optional.  Here its the `=` char that marks the argument as optional. 

As an alternative we could use the ES6 syntax in the actual function signature `, addrlen = 0`...but that adds some extra bytes to the output JS (maybe closure can remove this so its ok?).  For now this solution is what we've been going with.

Comment (kripken) [reply src/library.js:1878]: I see, thanks. sgtm as is.

File: src/preamble.js [hunk 2]
```diff
@@ -625,6 +625,14 @@ function abort(what) {
   // Use a wasm runtime error, because a JS error might be seen as a foreign
   // exception, which means we'd run destructors on it. We need the error to
   // simply make the program stop.
+
+  // Suppress closure compiler warning here. Closure compiler's builtin extern
+  // defintion for WebAssembly.RuntimeError claims it takes no arguments even
+  // though it can. For some rason our override in
```
Comment (kripken) [inline src/preamble.js:631]:
```suggestion
  // though it can. For some reason our override in
```

**Review by kripken — APPROVED** [review]

**Pull request merged**